profiling/energy-profiler/daemon.hpp

```diff
@@ -0,0 +1,43 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOSP_ENERGY_PROFILER_DAEMON_HPP
+#define KOKKOSP_ENERGY_PROFILER_DAEMON_HPP
+
+#include <chrono>
+#include <functional>
+#include <thread>
+
+namespace KokkosTools::EnergyProfiler {
+class Daemon {
+ public:
+  Daemon(std::function<void()> func, const std::chrono::duration& interval)
```

Comment:
One quick note is I don't see a default value for the time interval of the Daemon but maybe you could have one. If you do, I think a reasonable one would be 1 second. This would at least makes sense from the point of view of #291. Maybe there are other use cases and situations where you want it to be shorter (or longer).